lib/connections/quic_misc_test.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2019 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// +build go1.12
+
+package connections
+
+import (
+	"net"
+	"sort"
+	"testing"
+	"time"
+)
+
+type mockPacketConn struct {
+	addr net.Addr
+}
+
+func (mockPacketConn) ReadFrom(p []byte) (n int, addr net.Addr, err error) {
+	panic("implement me")
+}
+
+func (mockPacketConn) WriteTo(p []byte, addr net.Addr) (n int, err error) {
+	panic("implement me")
+}
+
+func (mockPacketConn) Close() error {
+	panic("implement me")
+}
+
+func (c *mockPacketConn) LocalAddr() net.Addr {
+	return c.addr
+}
+
+func (mockPacketConn) SetDeadline(t time.Time) error {
+	panic("implement me")
+}
+
+func (mockPacketConn) SetReadDeadline(t time.Time) error {
+	panic("implement me")
+}
+
+func (mockPacketConn) SetWriteDeadline(t time.Time) error {
+	panic("implement me")
+}
+
+func TestPacketConnUnspecifiedVsSpecified(t *testing.T) {
```

Comment:
This set of tests is tedious and not necessarily exhaustive, the whole file could be a table driven 

```
cases := []struct{
  netA string
  addrA string
  netB string
  addrB string
}{
  // a and b must be distinct, a is always the lesser one
  { "tcp", ":1235", "tcp", "127.0.0.1:1234" }, // unspecified is lesser
  ...
}

for _,tc := range cases {
  a, b := ... // resolve addresses, wrap in mockPacketConns
  if !packetConnLess(a, b) { fail }
  if packetConnLess(b, a) { fail }
  if packetConnLess(a, a) || packetConnLess(b, b) { fail }
}
```